### DISCUSSION TITLE
We can generate class names for `padding` and `margin`.

### DISCUSSION THREAD

**Discussion by iam-frankqiu** [tailwindlabs/tailwindcss#3082] [Ideas]
For example. When a user sets a class named as `p-98`. We can detect all the .html  .jsx  .vue .tsx files. Auto-generate these class names. What we only need to do is define the name rule. Finally. generating all the class. what do you think?

**Comment by simonswiss** [comment]
This is an interesting approach, but somewhat conflicts with the concept of "predefined constraints" where you set scales with specific values in the config file, and then stick to those values across the whole project.

Allowing users to define new values "on the fly" removes a lot of that design consistency.

A library called [Atomic CSS](https://acss.io), which has been around for a long time, takes that specific approach, if you're interested.

**Reply by iam-frankqiu** [reply]
> design

This approach is to solve a problem is that you can't list all the class names when it comes to `padding` and `margin`.
